### ISSUE TITLE
Final Project Peer Review

### ISSUE THREAD

**Issue by sukritipoddar** [Yazmine1213C/ORIE5741_Project#14]
The project is about classifying music genres using different lengths of audio clips as inputs that are converted into embedding matrices and plugged into a Graph Neural Network that classifies the music genres as both binary and then multiclass classification. Then various techniques are used to do feature selection and results are shared.

Strengths:
The report is very well structured with use binary classification and then build on it by using multiclass classification. A good number of results are shown with graphs showing the time (or epochs) the models take to converge. 
Since the project covers a lot of topics that have not been covered in class, the team has made a good effort to introduce the topics relevantly for readers and have given a good amount of explanation. 
The whole idea of the project is quite interesting and methodical. The team uses of PCA for dimensionality reduction, Control Burn for feature selection, KNN to determine the neighbours for nodes and employ an adjacency matrix to indicate the presence of an edge between nodes and the strength of an edge as well. This paints a well-rounded picture and gave a good idea of the flow of the project.
I like that the project has employed ControlBurn. It is interesting to see a novel concept in practice.

Areas of Improvement:
While the first part of the project is very well described, things seem a bit jumped in model evaluation section. Model Evaluation section mentions a move from PCA to TSNE(which is not explained what it is), changing Cosine similarity to other function which is not mentioned and different feature selection and subset. The improvements in results would have had a better intuitive understanding if these factors were discussed in detail with reasoning behind why the particular change is made to the model, for example, which function is replacing cosine similarity and why is it better; or which subset of feature is employed and why?
The activation function used for the Neural Nets is Relu. Since Relu does not bound its output, the input data needs to be normalized but the report does not discuss any normalization of input data to avoid getting very large values of weights. 
While the results are well described with good number of scores shown, an interpretation would help with the theory behind why the scores are achieved. For example, why increasing number of neighbours or using a particular distance metric changes the accuracy and how does it change the scores or convergence of the model.
Another factor that can be used in hyperparameter tuning is the embedding size of the embedding matrices. This can have tremendous effect on accuracy and the time it takes for the model to converge.
The report could also have a model selection section discussing the details of the final model that is concluded to be best for the genre classification.
Future prospects could use an ablation study after employing the above suggestions.

Overall, this is a promising project with intriguing academic prospects. I really enjoyed reading the report. Thanks!
